In [1]:
import os
import math
import pickle
import numpy as np
import pandas as pd

In [2]:
# 加载训练集、测试集数据
train = pd.read_csv('../data/train.csv')
test = pd.read_csv('../data/test.csv')

数据分布分析

In [9]:
max(train['uid']), len(np.unique(train['uid'])), \
max(train['iid']), len(np.unique(train['iid'])),

(223969, 157949, 14725, 14620)

In [10]:
max(test['uid']), len(np.unique(test['uid'])), \
max(test['iid']), len(np.unique(test['iid'])),

(223969, 142740, 14725, 12321)

训练集
    - uid
        - 最大值 223969
        - 有效值 157949
    - iid
        - 最大值 14725
        - 有效值 14620
测试集
    - uid
        - 最大值 223969
        - 有效值 142740
    - iid
        - 最大值 14725
        - 有效值 12321

由此可知，数据比较稀疏，uid、iid并非连续的。